src/main/frontend/handler/page.cljs

```diff
@@ -265,47 +265,80 @@
 
           (ok-handler))))))
 
+(defn rename-file!
+  [file new-name ok-handler]
+  (let [repo (state/get-current-repo)
+        old-path (:file/path file)
+        new-path (let [result (string/split old-path "/")
```

Comment:
It would be better to use some file system utilities/libraries.

Reply:
yes, we need to separate the `frontend.util` into multiple namespaces.